src/models/job.jsx

```diff
@@ -39,7 +39,7 @@ const JobContext = createContext({
 const updateJob = async (jobId, payload) => {
   // remove the jobId field if existed. No need to put the doc id into the doc data
   if (payload.id) {
-    payload.id == undefined;
+    payload.id = undefined;
```

Comment:
Sorry I should have just said this initially, objects passed to firebase [can't have undefined properties](https://stackoverflow.com/questions/34708566/firebase-update-failed-first-argument-contains-undefined-in-property) so maybe just use `delete` to remove the property?